nips/04.md

```diff
@@ -0,0 +1,55 @@
+NIP-04
+======
+
+Encrypted Direct Text Note
+--------------------------------------
+
+`draft` `optional`
+
+`kind` `4` `encrypted_direct_text_note`, is similiar to `kind` `1` `text_note`, but the content is encrypted and a tag targets a specific `public-key`.
+
+A shared cipher is generated by hashing the recipients `public-key` with the senders `private-key` 
+
+`content` is a base64 encoded, aes-256-cbc encrypted `string` of anything a user wants to write. 
+
+`tag` `p` is 32-byte hex of the recipents `public-key`. 
+
+In the first note `tag` `e` is a 32-byte sha256 hash of the shared cipher, so the recipient can validate the cipher they generate.
+
+Aditional messages include a `tag` `e` 32-byte hex of the message being replied to
+
```

Comment:
I'd like to see a more descriptive name for the tag names `e` and `p` to make the code more readable.
p could be pubkey
e could be reply_to

Reply:
I agree, but these two single-letter tags already have meaning defined on NIP-01, so we should change that there too, but then I'm too lazy to do it and update all the clients and relays.

For the next tags we create we should probably use more meaningful words.